test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
-        assert(isAround(delay, end()));
+      it('should throw an error while release a write session', function() {
+        let isSecondCall = false;
+        sessionPool.createTransaction_ = () => {
+          if (isSecondCall) {
+            return Promise.reject({});
+          }
+          isSecondCall = true;
+          return Promise.resolve();
+        };
+        sessionPool.session_ = function() {
+          return new Object({
+            create: () => Promise.resolve(),
+            delete: () => Promise.resolve(),
+          });
+        };
+        sessionPool.open();
+        clearTimeout(sessionPool.pingTimeoutHandle);
+        return sessionPool.getWriteSession().then(session =>
+          sessionPool.release(session).then(() => {
+            assert.strictEqual(sessionPool.getStats().writePool.available, 0);
+          })
+        );
       });
     });
-  });
 
-  describe('fill', function() {
-    var created;
+    describe('request', function() {
+      let CONFIG;
+      const SESSION = {
+        formattedName_: 'formatted-name',
+      };
 
-    beforeEach(function() {
-      sessionPool.options.min = 8;
+      beforeEach(function() {
+        CONFIG = {
+          reqOpts: {},
+        };
 
-      created = {
-        readonly: 0,
-        readwrite: 0,
-      };
+        sessionPool.getReadSession = function() {
+          return Promise.resolve(SESSION);
+        };
 
-      sessionPool.createSessionInBackground_ = function(type) {
-        created[type] += 1;
-      };
-    });
+        sessionPool.release = common.util.noop;
+      });
+
+      it('should get a session', function(done) {
+        sessionPool.getReadSession = function() {
+          setImmediate(done);
+          return Promise.resolve(SESSION);
+        };
 
-    it('should create the min number of required sessions', function() {
-      sessionPool.fill();
+        sessionPool.request(CONFIG, assert.ifError);
+      });
 
-      assert.strictEqual(created.readonly, 8);
-      assert.strictEqual(created.readwrite, 0);
-    });
+      it('should return error if it cannot get a session', function(done) {
+        const error = new Error('Error.');
 
-    it('should create the min number of write sessions', function() {
-      sessionPool.options.writes = 0.5;
-      sessionPool.fill();
+        sessionPool.getReadSession = function() {
+          return Promise.reject(error);
+        };
 
-      assert.strictEqual(created.readonly, 4);
-      assert.strictEqual(created.readwrite, 4);
-    });
+        sessionPool.request(CONFIG, function(err) {
+          assert.strictEqual(err, error);
+          done();
+        });
+      });
 
-    it('should respect the current size of the pool', function() {
-      sessionPool.options.writes = 0.5;
-      sessionPool.reads_ = [{}];
-      sessionPool.writes_ = [{}, {}];
-      sessionPool.fill();
+      it('should call the method with the session', function(done) {
```

Comment:
Can we update the description? It looks like we're just testing that the session name is applied to the request options.

Reply:
These test cases are not new. These are just copied from existing test cases